stdlib/public/core/StringObject.swift

```diff
@@ -498,8 +498,12 @@ extension _StringObject.CountAndFlags {
     self.init(zero:())
     self.count = count
     if isASCII {
+#if arch(i386) || arch(arm)
+      unimplemented_utf8_32bit()
+#else
       // ASCII also entails NFC
       self._storage |= 0xC000_0000_0000_0000
+#endif
```

Comment:
Since Karoy is currently working on 32-bit, drop this change so you don't conflict